src/main/java/dk/dbc/common/records/MarcRecordReader.java

```diff
@@ -34,7 +37,16 @@ public MarcRecordReader(MarcRecord marcRecord) {
      * @return The value of the subfield if found, <code>null</code> otherwise.
      */
     public String getValue(String fieldName, char subfieldName) {
-        return marcRecord.getSubFieldValue(fieldName, subfieldName).orElse(null);
+        final Optional<DataField> dataField = marcRecord.getField(DataField.class, hasTag(fieldName).and(hasSubField(subfieldName)));
+
+        if (dataField.isPresent()) {
+            final Optional<SubField> subField = dataField.get().getSubField(hasSubFieldCode(subfieldName));
+            if (subField.isPresent()) {
```

Comment:
Er det her ikke dobbeltkonfekt ? Det er jo konstateret i linie 40 at delfeltet er der.

Reply:
Jo, det virker lidt fjollet, men det er for ikke at få warnings pga Optional<>

Reply:
:-|